File: quivi.pos.react.app/src/hooks/api/Dtos/preparationgroups/PreparationGroup.ts
export interface PreparationGroup {
    readonly id: string;
    readonly sessionId: string;
    readonly isCommited: boolean;
    readonly note?: string;
    readonly orderIds: string[];
    readonly createdDate: string;
    readonly lastModified: string;
    readonly items: PreparationGroupItem[];
}

export interface BasePreparationGroupItem {
    readonly id: string;
    readonly menuItemId: string;
    readonly remainingQuantity: number;
    readonly quantity: number;
    readonly locationId?: string;
}

export interface PreparationGroupItem extends BasePreparationGroupItem {
    readonly extras: BasePreparationGroupItem[];
}

File: Quivi.Pos.Api/Dtos/PreparationGroup.cs
namespace Quivi.Pos.Api.Dtos
{
    public class PreparationGroup
    {
        public required string Id { get; init; }
        public required string SessionId { get; init; }
        public bool IsCommited { get; init; }
        public string? Note { get; init; }
        public required IEnumerable<string> OrderIds { get; init; }
        public required IEnumerable<PreparationGroupItem> Items { get; init; }
        public DateTimeOffset CreatedDate { get; init; }
        public DateTimeOffset LastModified { get; init; }
    }
}